consensus/src/dag/bootstrap.rs

```diff
@@ -164,40 +165,89 @@ impl TDagMode for SyncMode {
-            Mode::Active(ActiveMode {
-                handler: new_handler,
-                fetch_service: new_fetch_service,
-                base_state: self.base_state,
-            })
+        let commit_li = self.certified_node_msg.ledger_info().clone();
+
+        let network_handle = SyncModeMessageHandler::new(
+            bootstrapper.epoch_state.clone(),
+            request.start_round(),
+            request.target_round(),
+        );
+
+        let (shutdown_handle, shutdown) = shutdown_handle.new_child();
+
+        let (res_tx, res_rx) = oneshot::channel();
+        tokio::spawn(async move {
+            let success = match sync_manager
+                .sync_dag_to(
+                    dag_fetcher,
+                    request,
+                    responders,
+                    sync_dag_store,
+                    commit_li,
+                    shutdown,
+                )
+                .await
+            {
+                Ok(_) => {
+                    info!("sync success. going to rebootstrap.");
+                    true
+                },
+                Err(_) => {
+                    info!("sync failed. continuing without advancing.");
+                    false
+                },
+            };
+            let _ = res_tx.send(success);
+        });
+
+        let mut buffer = Vec::new();
+
+        select! {
+            res = network_handle.run(dag_rpc_rx, &mut buffer) => {
+                // The network handle returns if the sender side of dag_rpc_rx closes,
+                // or network handle found a future CertifiedNodeMessage to cancel the
+                // current sync.
+                if let Some(msg) = res {
+                    shutdown_handle.shutdown().await;
+                    return Mode::Sync(SyncMode {
+                        certified_node_msg: msg,
+                        base_state: self.base_state,
+                    });
+                } else {
+                    // Usually happens if the sender side of dag_rpc_rx closes first, which
+                    // shouldn't happen.
+                    return Mode::Exit(ExitMode {  })
+                }
+            },
+            res = res_rx => {
+                if let Ok(true) = res {
+                    // If the sync task finishes successfully, we can transition to Active mode by
+                    // rebootstrapping all components starting from the DAG store.
+                    let (new_state, new_handler, new_fetch_service) = bootstrapper.full_bootstrap();
+                    return Mode::Active(ActiveMode {
+                        handler: new_handler,
+                        fetch_service: new_fetch_service,
+                        base_state: new_state,
+                        buffer,
+                    })
+                } else {
+                    // If the sync task fails, then continue the DAG in Active Mode with existing state.
```

Comment:
+1, it seems continuing in sync mode is simpler? although not sure if there's edge case that malicious node can trick you into sync mode and no other honest nodes can. i.e. every honest nodes is at the window - 1 to trigger your state sync, but malicious has the one beyond the window

Reply:
Based on offline discussion, we will leave this as is. We can revisit if our testing reveals anything.